### DISCUSSION TITLE
Can't update via docker-compose on Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by andyross** [teslamate-org/teslamate#1745] [Getting started]
Seems like teslamate addresses version hell well enough for its own software but not for its build dependencies?  When I try the instructions for updating via "docker-compose pull", it tosses an error about a missing symbol in the six library:

    AttributeError: module 'six' has no attribute 'ensure_text'

As it happens, six is installed via the distro package and is version 1.14.0.  I see the upstream latest calls itself 1.16.0.  docker-compose was installed via pip.

What's the correct recipe for untangling this?

**Comment by andyross** [comment]
Came back to struggle with this some.  Short of using a venv, there really isn't much to be done.  I just carefully removed the existing package without touching things that depended on it and re-installed "six" via pip.  Nothing seems to have broken.

It's probably worth revisiting the install instructions though, given that docker-compose doesn't actually run on the current LTS version of the world's most popular distro.

**Comment by DrMichael** [comment]
I have docker-compose running under Ubuntu 20.04.2 with version 1.14.0-2 without issues.

Might be an issue with python-six vs. python3-six. Check, what you have installed with `apt --installed list | grep six | grep python` , I have python3-six/focal.
